### PR TITLE
DATACOUCH-383 - RxJavaCouchbaseTemplate should return generated id wh…

### PR DESCRIPTION
**Pull request by davidkelly**
…en saving.

Seems we fixed this for the non-Rx version, but not the RxJava version.  Followed
same basic strategy which was used for version, added a simple test.

### CODE REVIEW

File: src/test/java/org/springframework/data/couchbase/repository/ReactivePlace.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.springframework.data.couchbase.repository;
+
+import com.couchbase.client.java.repository.annotation.Field;
+import com.couchbase.client.java.repository.annotation.Id;
+import org.springframework.data.couchbase.core.mapping.id.GeneratedValue;
+import org.springframework.data.couchbase.core.mapping.id.GenerationStrategy;
+
+/** author
+ * David Kelly
```
Comment (dnault) [inline src/test/java/org/springframework/data/couchbase/repository/ReactivePlace.java]: Did you mean for this to be a Javadoc "author" tag?

    /**
      * @author David Kelly
      */

File: src/test/java/org/springframework/data/couchbase/repository/ReactivePlaceRepository.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.springframework.data.couchbase.core.query.N1qlPrimaryIndexed;
+import org.springframework.data.couchbase.core.query.Query;
+import org.springframework.data.couchbase.core.query.ViewIndexed;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+
+/** author
+ * David Kelly
```
Comment (dnault) [inline src/test/java/org/springframework/data/couchbase/repository/ReactivePlaceRepository.java]: Ditto, should probably be Javadoc tag

File: src/test/java/org/springframework/data/couchbase/repository/ReactivePlaceTests.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.data.repository.core.support.ReactiveRepositoryFactorySupport;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+
+import static junit.framework.TestCase.assertNull;
+import static org.junit.Assert.assertNotNull;
+import static org.springframework.data.couchbase.CouchbaseTestHelper.getRepositoryWithRetry;
+
+/**  author
+ * David Kelly
```
Comment (dnault) [inline src/test/java/org/springframework/data/couchbase/repository/ReactivePlaceTests.java]: Ditto, should probably be Javadoc tag

File: src/test/java/org/springframework/data/couchbase/repository/ReactivePlace.java (hunk 1 above)
Comment (davidkelly) [reply src/test/java/org/springframework/data/couchbase/repository/ReactivePlace.java]: Doh!  Thx for catching that!

**Pull request merged**